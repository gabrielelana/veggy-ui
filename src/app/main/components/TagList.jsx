function TagList({tags}){
  const rows = tags.map(t => (<tr key={t.tag}>
    <td>{t.tag}</td>
    <td>{t.pomodori}</td>
    </tr>))
  return (
    <div className="column is-one-third">
      <table className="table">
        <thead>
          <tr>
            <th>Tag</th>
            <th>Pomodori</th>
          </tr>
        </thead>
        <tbody>
         {rows}
        </tbody>
      </table>
    </div>
  )
}


export default TagList