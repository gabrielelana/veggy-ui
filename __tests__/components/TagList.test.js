import React from 'react'
import renderer from 'react-test-renderer'
import TagList from '../../src/app/main/components/TagList'

describe('<TagList />', () => {
  
  it('should display tags', () => {
    const tags = [
      {tag: '#foo', pomodori: 4},
      {tag: '#bar', pomodori: 7},
      ]
    const component = renderer.create(<TagList tags={tags} />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})