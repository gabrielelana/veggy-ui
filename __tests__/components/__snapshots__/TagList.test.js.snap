// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagList /> should display tags 1`] = `
<div
  className="column is-one-third"
>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Tag
        </th>
        <th>
          Pomodori
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          #foo
        </td>
        <td>
          4
        </td>
      </tr>
      <tr>
        <td>
          #bar
        </td>
        <td>
          7
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
